File: kotlin/hamming/src/main/kotlin/Hamming.kt
object Hamming {
    fun compute(a: String, b:String): Int {
        if (a.length != b.length) {
            throw IllegalArgumentException("left and right strands must be of equal length.")
        }

        return a.zip(b).count { pair -> pair.first != pair.second }
    }
}

File: cpp/hamming/hamming.cpp
#include <stdexcept>
#include "hamming.h"

namespace hamming {

int compute(std::string a, std::string b) {
    if (a.length() != b.length()) {
        throw std::domain_error(std::string("Strings must be the same size"));
    }

    int sum = 0;

    for (int i = 0; i <= a.length(); ++i) {
        if (a[i] != b[i]) {
            ++sum;
        }
    }

    return sum;
}

} // namespace hamming
